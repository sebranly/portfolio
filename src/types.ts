// TODO: have a glossary on the website
// TODO: explain myProjects series
export enum Tag {
  All = 'All',
  ThreeDimensional = '3D',
  AIML = 'AIML',
  Archived = 'archived',
  C = 'C',
  CSS = 'CSS',
  CLI = 'CLI',
  DNS = 'DNS',
  GitHub = 'GitHub',
  GUI = 'GUI',
  HTML = 'HTML',
  i18n = 'i18n',
  JavaScript = 'JavaScript',
  Live = 'live',
  MyProjects = 'myProjects',
  NintendoDS = 'Nintendo DS',
  OpenGL = 'OpenGL',
  Puzzle = 'puzzle',
  Python = 'Python',
  React = 'React',
  SchoolProject = 'school project',
  SDL = 'SDL',
  VideoGame = 'video game',
  Website = 'website',
  PAlib = 'PAlib',
  PHP = 'PHP',
  SQL = 'SQL',
  TailwindCSS = 'TailwindCSS',
  TypeScript = 'TypeScript',
  Windows = 'Windows'
}

export enum Color {
  Black = 'black',
  White = 'white',
  Gray = 'gray',
  Red = 'red',
  Orange = 'orange',
  Amber = 'amber',
  Yellow = 'yellow',
  Green = 'green',
  Blue = 'blue',
  Indigo = 'indigo',
  Purple = 'purple',
  Pink = 'pink',
  Rose = 'rose'
}

export enum Role {
  Algorithm = 'Algorithm',
  Developer = 'Developer',
  Designer = 'Designer',
  ProductOwner = 'Product Owner',
  Tester = 'Tester'
}

export type Contributor = {
  female?: boolean;
  name: string;
  roles: Role[];
};

export type Project = {
  download?: string;
  title: string;
  contributors?: Contributor[];
  github?: string;
  subtitle: string;
  description: string[];
  tags: Tag[];
  video?: string;
  website?: string;
  years: number[];
  textualSnapshot?: string[];
};

export type Language = {
  shortName: string;
  name: string;
};
