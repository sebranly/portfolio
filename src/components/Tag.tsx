import * as React from 'react';
import { Color, Tag as TagType } from '../types';
import classnames from 'classnames';
import { getTagColor, getTailwindBackgroundColor, getTailwindHoverBackgroundColor, hasTagTranslation } from '../utils';
import { useTranslation } from 'react-i18next';

export interface TagProps {
  tag: TagType | number;
}

const Tag: React.FC<TagProps> = (props) => {
  const { t } = useTranslation();
  const { tag } = props;
  const color = getTagColor(tag);
  const tagText = hasTagTranslation(tag) ? t(`projects.general.tags.${tag}`) : tag;
  const isWhite = color === Color.White;

  const classnamesTag = classnames(
    'border-2 border-black border-solid cursor-pointer inline-block rounded mr-2 my-1 px-4 py-1',
    { 'text-white': !isWhite, 'text-black': isWhite },
    getTailwindBackgroundColor(color),
    getTailwindHoverBackgroundColor(color)
  );

  return <div className={classnamesTag}>{tagText}</div>;
};

export { Tag };
