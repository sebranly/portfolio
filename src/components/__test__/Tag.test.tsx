import * as React from 'react';
import { render } from '@testing-library/react';
import { Tag as TagType } from '../../types';
import { Tag } from '../Tag';

test('Tag', () => {
  const tag = TagType.JavaScript;
  const createComponent = () => <Tag tag={tag} />;

  const { container } = render(createComponent());
  expect(container.childNodes).toMatchSnapshot();
});
