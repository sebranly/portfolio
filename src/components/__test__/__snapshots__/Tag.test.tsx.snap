// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag 1`] = `
NodeList [
  <div
    class="border-2 border-black border-solid inline-block rounded mr-2 my-1 px-4 py-1 text-white bg-yellow-600 hover:bg-yellow-700"
  >
    JavaScript
  </div>,
]
`;
